File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.py
import heapq
def solution(operations):
    answer = []
    min_q=[]
    max_q=[]
    l=0
    for cur in operations:
        s=cur.split()
        op=s[0]
        num=int(s[1])
        if op=='I':
            heapq.heappush(min_q,num)
            heapq.heappush(max_q,-num)
            l+=1
        elif op=='D':
            if l:
                if num==1:
                    heapq.heappop(max_q)
                else:
                    heapq.heappop(min_q)
                l-=1
            if not l:
                min_q.clear()
                max_q.clear()
    a=0
    b=0
    if min_q:
        b=heapq.heappop(min_q)
    if max_q:
        a=heapq.heappop(max_q)
    return [-a,b]

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.cpp
#include <bits/stdc++.h>

using namespace std;
multiset<int>mq;
vector<int> solution(vector<string> operations) {
    vector<int> answer;
    for (auto cur : operations)
    {
        char op;
        op=cur[0];
        string s = cur.substr(2);
        int num = stoi(s);
        if (op=='I')
        {
            mq.insert(num);
        }
        else if (op=='D' && mq.size())
        {
            if (num==-1){
                mq.erase(mq.begin());
            }
            else{
                mq.erase(--mq.end());
            }
        }
    }
    if (mq.size()>1)
        return {*--mq.end(),*mq.begin()};
    return {0,0};
}